api/http/api_deployments.go

```diff
@@ -891,50 +1045,38 @@ func (d *DeploymentsApiHandlers) GetDeploymentForDevice(w rest.ResponseWriter, r
 	if deployment == nil {
 		d.view.RenderNoUpdateForDevice(w)
 		return
+	} else if deployment.Type == model.DeploymentTypeConfiguration {
+		// Generate pre-signed URL
+		var hostName string
+		if hostName = r.Header.Get("X-Forwarded-Host"); hostName == "" {
```

Comment:
very good idea on having the presign host, and even the presign scheme. you could say we both honor the X-Forwarded header, and have a configuration fallback, right?

I think I would revert the logic here though. what if we promote the config setting:
- if the setting is there - use that
- if the setting is not there - use X-Forwarded-Host
- otherwise, fail

 the reason is I'm having second thoughts. I know I cofirmed that we can trust X-Forwarded-Host, because I'm pretty sure we can trust traefik to set it. I don't know however if in a live aws setup, with it's load balancer, we will get the original DNS name of the gateway here.
 
this will be researched and POCed still, so just a thought.

Reply:
Fair point, I'll change the logic to prefer the configuration setting and blank out the default config value.